dashboard/src/components/AppUpgrade/AppUpgrade.tsx

```diff
@@ -93,7 +94,7 @@ class AppUpgrade extends React.Component<IAppUpgradeProps, IAppUpgradeState> {
         repo &&
         repo.metadata &&
         repo.metadata.name &&
-        (prevProps.app !== app || prevProps.repo !== repo)
+        (prevProps.app !== app || prevRepo !== repo)
```

Comment:
I had expected the test you added with 7b966848d to fail with this change, as it was added to verify that if the props.repo changes it fetches the data? Why is it not failing?

Reply:
In that test, `state.repo` is empty at the beginning. Then component gets updated with `props.repo`.  `componentDidUpdate` gets executed and the state gets updated. Since `prevRepo !== repo` the function gets called.